modules/@apostrophecms/asset/lib/webpack/utils.js

```diff
@@ -0,0 +1,47 @@
+const { merge: webpackMerge } = require('webpack-merge');
+
+module.exports = {
+  checkModulesWebpackConfig(modules, t) {
+    const allowedProperties = [ 'extensions', 'bundles' ];
+
+    for (const mod of Object.values(modules)) {
+      const webpackConfig = mod.__meta.webpack[mod.__meta.name];
+
+      if (!webpackConfig) {
+        continue;
+      }
+
+      if (
+        typeof webpackConfig !== 'object' ||
+        webpackConfig === null ||
+        Array.isArray(webpackConfig) ||
+        Object.keys(webpackConfig).some((prop) => !allowedProperties.includes(prop))
+      ) {
+        const error = t('apostrophe:assetWebpackConfigWarning', {
+          module: mod.__meta.name
+        });
+
+        throw new Error(error);
+      }
+    }
+  },
+  mergeWebpackConfigs (modules, config) {
+    const extensions = Object.values(modules).reduce((acc, mod) => {
+      const { name, webpack } = mod.__meta;
```

Comment:
At this point, `webpack` could have a subproperty for every subclass that makes up the inheritance tree for `name`. That's intentional and we want all of them. So your merge should walk through all of them in order, rather than taking only `webpack[name]`.

Otherwise a base class and a subclass cannot contribute separate webpack extensions, which is allowed.